ui/components/app/confirm/scroll-to-bottom/scroll-to-bottom.tsx

```diff
@@ -0,0 +1,144 @@
+import React, {
+  Dispatch,
+  SetStateAction,
+  useContext,
+  useEffect,
+  useState,
+} from 'react';
+import { debounce } from 'lodash';
+import { I18nContext } from '../../../../contexts/i18n';
+import {
+  Box,
+  ButtonIcon,
+  ButtonIconSize,
+  IconName,
+} from '../../../component-library';
+import type { StyleUtilityProps } from '../../../component-library/box';
+
+import {
+  BackgroundColor,
+  BlockSize,
+  Display,
+  FlexDirection,
+  IconColor,
+  BorderRadius,
+} from '../../../../helpers/constants/design-system';
+
+interface ContentProps extends StyleUtilityProps {
+  /**
+   * Elements that go in the page content section
+   */
+  children: React.ReactNode | React.ReactNode[];
+  /**
+   * Is true when all content has been displayed
+   */
+  hasViewedContent?: boolean;
+  /**
+   * Setter function for hasViewedContent
+   */
+  setHasViewedContent?: Dispatch<SetStateAction<boolean>>;
+}
+
+const ScrollToBottom = ({
+  children,
+  hasViewedContent,
+  setHasViewedContent,
+  ...props
+}: ContentProps) => {
+  const t = useContext(I18nContext);
+  const [showScrollDown, setShowScrollDown] = useState(false);
+
+  const containerRef = React.createRef<HTMLSpanElement>();
+  const bottomRef = React.createRef<HTMLDivElement>();
+
+  const handleDebouncedScroll = debounce((target) => {
```

Comment:
I don't have all the context here, but I wrote a hook for a lot of this a while back. Wondering if we can reduce some duplication by using it? 🤔 

https://github.com/MetaMask/metamask-extension/blob/develop/ui/hooks/useScrollRequired.js

Reply:
@FrederikBolding 

yo... waht. this is great. ok, I made updates to use this hook. I added a few things to the hook. let me know what you think 

1. added smooth scroll behavior. also removed x scroll pos since I don't think it's needed
2. added hasScrolledToBottomState
  - with the current hook implementation, it seemed that after scrolling to the bottom and then scrolling back up, the scroll button would no longer be visible. "hasScrolledToBottomState" is used in replace of the former "isScrolledToBottomState". "isScrolledToBottomState" now represents when we are scrolled to the bottom.
3. minor update to fix TS useRef() → useRef(null)

I like how you made this into a hook, didn't need to use a bottomRef (an oversight in my earlier implementation), and have a buffer for isScrolledToBottom. thanks for this comment!